src/main/app/forms/validation/validators/atLeastOneFieldIsPopulated.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface
+} from 'class-validator'
+
+@ValidatorConstraint()
+export class AtLeastOneFieldIsPopulatedConstraint implements ValidatorConstraintInterface {
+  validate (value: any, args?: ValidationArguments) {
+    if (value === undefined) {
+      return true
+    }
+
+    return Object.keys(value).some(key => !!value[key])
+  }
+}
+
+/**
+ * Validates if one of the fields in object is populated. It doesn't matter which one as long as one
+ * as long as at least one of them has any value. It doesn't matter what is the value as long as it's no falsy.
```

Comment:
This documentation does not read well :(

Reply:
reworded. Yeah, this one must have been written by drunken bear ;)